### ISSUE TITLE
`BulkMessageResponse` doesn't handle `Vec<Message>`

### ISSUE THREAD

**Issue by vloddot** [toastxc/Reywen-Revolt#44]
![image](https://github.com/vloddot/jolt/assets/97629148/00f1abc4-196b-4c39-8b20-3efceb1f4905)
Essentially, Revolt servers respond with a `Vec<Message>` instead of an expected struct of
```rust
{
    messages: Option<Vec<Message>>,
    users: Option<Vec<User>>,
    members: Option<Vec<Member>>,
}
```
This usually happens in small/new channels. This issue has been addressed before but has never been fixed.
For reproduction, this usually happens in servers like yui's server with most channels.

Try out different servers as well, or maybe create a testing server and just create a new channel. It usually happens there.

The channels #Joins / Leaves and #Testing in the video are mostly empty. #Joins / Leaves only has a few messages and #Testing has no messages.

https://github.com/toastxc/Reywen-Revolt/assets/97629148/13f4dc94-4cc1-4992-a27a-cbef0c34bb2d

**Comment by vloddot** [comment]
With some more testing, adding YouTube embeds apparently changes whether the server gives a `Vec<Message>` or not. Very weird behavior

https://github.com/toastxc/Reywen-Revolt/assets/97629148/49e8e694-3cc2-45a7-8eb1-be586665ff5b

**Comment by toastxc** [comment]
that is indeed weird, embeds affect the fields on the `Message` structure but should not affect the enum varient. as the varients are based on the channel type.

**Comment by vloddot** [comment]
This issue is fixed in the latest 0.1.23 release. Thanks @toastxc!
